feat: Add no-unexpected-wire-adapter-usages linting rule (#30)
salesforce/eslint-plugin-lwc: docs/rules/no-unexpected-wire-adapter-usages.md

## Enforce wire adapters to be used with `wire` decorator (no-unexpected-wire-adapter-usages)

> This rule offers a way to enforce that wire adapters are only used in the `wire` decorator. It prevents imperative invocation, reassignment, and so on.

Reviewer: ```suggestion
This rule prevents unexpected wire adapter usage by ensuring that they are only used via the `wire` decorator.
```

Reviewer 2: Great suggestion @ekashida

> This rule offers a way to enforce that wire adapters are using along with `wire` decorator. This prevents for example wire adapter: imperative invocation, reassignment, ...

Reviewer 2: ```suggestion
This rule offers a way to enforce that wire adapters are only used in the `wire` decorator. It prevents imperative invocation, reassignment, and so on.
```

### Rule details

> This rule can be configured as followed:

Reviewer 2: ```suggestion
This rule can be configured as follows:
```
